src/plugins/workspace/public/components/workspace_picker_content/workspace_picker_content.tsx

```diff
@@ -88,24 +165,63 @@ export const WorkspacePickerContent = ({
       );
     });
     return (
-      <EuiListGroup showToolTips flush gutterSize="none" wrapText maxWidth={240}>
-        <EuiListGroupItem
-          label={
-            <EuiTitle size="xxs">
-              <h4>{itemType === 'all' ? allWorkspacesTitle : recentWorkspacesTitle}</h4>
-            </EuiTitle>
-          }
-        />
-        {listItems}
-      </EuiListGroup>
+      <>
+        <EuiTitle size="xxs">
+          <h4>{itemType === 'all' ? allWorkspacesTitle : recentWorkspacesTitle}</h4>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiListGroup showToolTips flush gutterSize="none" wrapText maxWidth={240}>
+          {listItems}
+        </EuiListGroup>
+        <EuiSpacer size="s" />
+      </>
     );
   };
 
   return (
     <>
-      {filteredRecentWorkspaces.length > 0 &&
-        getWorkspaceListGroup(filteredRecentWorkspaces, 'recent')}
-      {filteredWorkspaceList.length > 0 && getWorkspaceListGroup(filteredWorkspaceList, 'all')}
+      <EuiFlexGroup
+        direction="column"
+        justifyContent="center"
+        alignItems="center"
+        style={{ height: '100%' }}
+        gutterSize="none"
+      >
+        <EuiFlexItem grow={false} style={{ position: 'sticky', width: '100%' }}>
+          <EuiFieldSearch
+            compressed={true}
+            fullWidth={true}
+            value={search}
+            onChange={(e) => handleSearchInput({ text: e.target.value })}
+            placeholder={searchFieldPlaceholder}
+          />
+          <EuiSpacer />
+        </EuiFlexItem>
+
+        <EuiFlexItem grow={true} style={{ width: '100%', overflowY: 'auto' }}>
+          <EuiPanel paddingSize="none" color="transparent" hasBorder={false}>
+            {search ? (
+              queriedWorkspace && queriedWorkspace.length > 0 ? (
+                <>
+                  {filteredRecentWorkspaces.length > 0 &&
+                    getWorkspaceListGroup(queriedRecentWorkspace, 'recent')}
+                  {workspaceList.length > 0 && getWorkspaceListGroup(queriedWorkspace, 'all')}
+                </>
+              ) : (
+                getEmptyStatePrompt()
+              )
+            ) : (
+              <>
+                {filteredRecentWorkspaces.length > 0 &&
+                  getWorkspaceListGroup(filteredRecentWorkspaces, 'recent')}
+
+                {workspaceList.length > 0 && getWorkspaceListGroup(workspaceList, 'all')}
+                {workspaceList.length === 0 && <>{getEmptyStatePrompt()}</>}
+              </>
+            )}
```

Comment:
```suggestion
            {filteredRecentWorkspaces.length > 0 &&
                  getWorkspaceListGroup(filteredRecentWorkspaces, 'recent')}
            {queriedWorkspace.length > 0 &&
                  getWorkspaceListGroup(queriedWorkspace, 'all')}
            {queriedWorkspace.length === 0 && getEmptyStatePrompt()}
```
The logic here is a little bit complex for me, can we change to like this?

Reply:
Thank you for the comments, I have updated them.